### PR TITLE
Expose LSF_RESOURCES and LSF_SERVER in keys

### PR DESCRIPTION
**Pull request by pgdr**
We expose two more LSF related keys in `ConfigKeys`
* `LSF_RESOURCES_KEY`
* `LSF_SERVER_KEY`

There was some internal confusion regarding `queue_name` and `queue_system`, that is, the functions named `queue_name` were used to get the `queue_system` information (e.g. LSF/LOCAL/TORQUE), but this has now been changed to `queue_system`.

Furthermore, `LSF_RESOURCE` was named `LSF_RESOURCES` several places, and one place, something is called `LSF_REQUEST`.  The _actual IBM concept_ is called **Resource requirements**.

I've tried to change most of them to `LSF_RESOURCE`.

### CODE REVIEW

**Review by joakim-hove — DISMISSED** [review]

**Comment by pgdr** [comment]
Ready for review

File: lib/include/ert/enkf/config_keys.h [hunk 1]
```diff
@@ -102,8 +102,8 @@ extern "C" {
 #define  LOCAL_CONFIG_KEY                  "LOCAL_CONFIG"
 #define  LOG_FILE_KEY                      "LOG_FILE"
 #define  LOG_LEVEL_KEY                     "LOG_LEVEL"
-#define  LSF_QUEUE_KEY                     "LSF_QUEUE"
-#define  LSF_RESOURCES_KEY                 "LSF_RESOURCES"
+#define  LSF_QUEUE_KEY                     "LSF_QUEUE" // The queue name, e.g. "mr"
```
Comment (joakim-hove) [inline lib/include/ert/enkf/config_keys.h:105]: Sorry - I suggest these keys are left unmodified; maybe with a deprecation warning. The keys which are actually accessed through the dictionary based api - and hence everest are [these keys](https://github.com/Statoil/libres/blob/master/lib/include/ert/job_queue/lsf_driver.h#L32).

File: python/res/enkf/queue_config.py [hunk 2]
```diff
@@ -57,12 +58,25 @@ def max_submit(self):
...
     def driver(self):
-        return self._queue_driver(self.queue_name).setParent(self)
+        return self._queue_driver(self.queue_system).setParent(self)
 
     @property
     def job_script(self):
         return self._get_job_script()
+
+    @property
+    def lsf_resource(self):
```
Comment (joakim-hove) [inline python/res/enkf/queue_config.py]: This should in my opinion explicitly ask for the LSF driver, and then return resource. The following situation is realistic:

1. The user has configured to use the `LOCAL` system.
2. When asking for the `lsf_resource` the queue config will look up the active - i.e. "LOCAL" system, and then query that for lsf resource -> Exception.

**Comment by joakim-hove** [comment]
I realize this is quite messy in the current situation, and appreciate the cleanup resulting from your effort.

**Comment by pgdr** [comment]
Thanks for the review, I had to fix a bug first, and now I will address your comments.

File: python/res/enkf/queue_config.py (hunk 2 above)
Comment (pgdr) [reply python/res/enkf/queue_config.py]: done

File: lib/include/ert/enkf/config_keys.h (hunk 1 above)
Comment (pgdr) [reply lib/include/ert/enkf/config_keys.h:105]: I think that's done now?  Can you check and see if this is satisfactory?  (Perhaps I've misunderstood something, though.)

**Review by joakim-hove — APPROVED** [review]

**Pull request merged**